imap_processing/hit/l1a/hit_l1a.py

```diff
@@ -308,25 +335,268 @@ def add_cdf_attributes(
     return dataset
 
 
+def find_complete_mod10_sets(mod_vals: np.ndarray) -> np.ndarray:
+    """
+    Find start indices where mod_vals matches [0,1,...,9] pattern.
+
+    Parameters
+    ----------
+    mod_vals : np.ndarray
+        1D array of mod 10 values from the hdr_minute_cnt field in the L1A counts data.
+
+    Returns
+    -------
+    np.ndarray
+        Indices in mod_vals where the complete pattern [0, 1, ..., 9] starts.
+    """
+    window_size = len(MOD_10_PATTERN)
+
+    if mod_vals.size < window_size:
+        logger.warning(
+            "Mod 10 array is smaller than the required window size for "
+            "pattern matching."
+        )
+        return np.array([], dtype=int)
+    # Use sliding windows to find pattern matches
+    sw_view = np.lib.stride_tricks.sliding_window_view(mod_vals, window_size)
+    matches = np.all(sw_view == MOD_10_PATTERN, axis=1)
```

Comment:
I find the MOD_10_PATTERN a bit confusing because it is really just a continuous range of numbers it looks like. What about doing directly the arange() call here? where window_size = 10 would be defined above.
```suggestion
    matches = np.all(sw_view == np.arange(window_size), axis=1)
```